### ISSUE TITLE
How can I test the model on single image dataset?

### ISSUE THREAD

**Issue by m1spivot** [Tangshitao/Dense-Scene-Matching#13]
Hi, I noticed that, in the paper, DSM shows 2 different performance respectively on so called **Single** and **Video**.

I am trying to repeat your work, but it seems that the _dataset.py_ only provides video data. It works fine and I have got correct performance after video test.

Now I wonder how can I test the model on single image dataset? (using 7scenes)

**Comment by Tangshitao** [comment]
Hi, I think you can remove the temporal fusion modules in https://github.com/Tangshitao/Dense-Scene-Matching/blob/ccbfc7aedbf0a54b5120cc4fd0e2a3f72238f027/libs/model/arch/DSMNet.py#L410

**Comment by m1spivot** [comment]
Thanks a lot, it works!

I have another question, that if I want to add some layers (like a fc layer or something else) in FPN, will weights of new layers be saved like those in **_model_best.pht_**?

**Comment by Tangshitao** [comment]
Yes

**Comment by m1spivot** [comment]
I trained the model after adding some layers, now the size of **checkpoint.pth** is 1.3GB.

But when I run **video_test.py** loading my new **checkpoint.pth**, it is extremely slow so that one batch may spends 2-3 minutes!! The console's printing is "Sampling 256 hypotheses.". By the way, I checked hardware usage and it showed that CPU 100% running while GPU didn't.

I do not know what causes this. Is my model too big? Or are these some bugs may make it running on CPU rather than GPU?

**Comment by Tangshitao** [comment]
It think the additional layers broken the optimization, so the predicted coordinates are not accurate. The PnP solver becomes slow if the coordinates are noisy. You can check the accuracy of the coordinates.

**Comment by m1spivot** [comment]
I tested my modified model on only 20 batch and it was awful that translation error was 4.36 while rotation error was 126.16. How to calculate the accuracy of coordinates so that I can optimize it?
![image](https://user-images.githubusercontent.com/46309122/236528177-cd70b677-a74f-406c-b83d-40c06c4e8833.png)

**Comment by Tangshitao** [comment]
Compute the coordinate errors between predicted coordinates and ground truth coordinates.

**Comment by m1spivot** [comment]
Thank you for the answer. I am training my modified model on 7scene dataset now!

There are some coordinates image comparison where the left ones are my prediction, the middle ones are origin DSM prediction and the right ones are ground truth coordinates.

It can be seen that my prediction is not so good and the coordinates loss has stuck at about 0.2 for a long time. Total loss fluctuates between about 0.4 and 1.2 The learning rate now is 0.0001. I have no idea why this happened. Should I choose a smaller learning rate? Will using reproj_loss work to optimize?

![屏幕截图 2023-05-06 112341](https://user-images.githubusercontent.com/46309122/236691054-484bce14-7b20-47b6-842a-db572571239b.png)
![屏幕截图 2023-05-06 114814](https://user-images.githubusercontent.com/46309122/236691062-e340e0c9-e9c8-4623-87cc-a0cb8b35a35c.png)
![屏幕截图 2023-05-06 115452](https://user-images.githubusercontent.com/46309122/236691066-c76dd175-53e7-4f80-a286-e1373f8bfcf5.png)
